packages/core/test/unit/fs/project-file.spec.ts

```diff
@@ -240,6 +242,99 @@ describe(ProjectFile.name, () => {
     });
   });
 
+  describe('chmod', () => {
```

Comment:
I see you've separated the mode tests into a `describe` here. This is not what I would prefer. I would assume there to be a `describe` per public method. 

In this case, I would expect 1~ish additional test per public method to test the `chmod` stuff specifically. 

What do you think?